core/darwin/src/internal/TimeZoneNative.kt

```diff
@@ -12,10 +12,18 @@ import kotlinx.datetime.internal.*
 import platform.Foundation.*
 
 internal actual fun timeZoneById(zoneId: String): TimeZone =
-    RegionTimeZone(tzdb.getOrThrow().rulesForId(zoneId), zoneId)
+    runCatching { RegionTimeZone(tzdb.getOrThrow().rulesForId(zoneId), zoneId) }
+        .getOrElse { timeZoneByIdFoundation(zoneId) }
```

Comment:
With this implementations, the system can enter a weird state where one part of the data comes from the filesystem, whereas the other comes from the fallback. I think it makes more sense to only check once whether to use a filesystem or the fallback, and then consistently stick to that.